### DISCUSSION TITLE
expand and collapse all rows in a dataTable

### DISCUSSION THREAD

**Discussion by gunnarschmid** [primefaces/community#1266] [PrimeFaces]
Hi!
Is there a standard way to expand and collapse all rows in a p:dataTable?
The showcase does not provide an example, and all I find on SO is some Javascript snippets.
I tried to toggle the expandedRows flag via a commandLink (in the header facet), but this gets messed up when individual rows are expanded manually.

**Comment by melloware** [comment]
Well the widget has a `collapseAllRows()` method https://primefaces.github.io/primefaces/13_0_0/#/components/datatable?id=client-side-api

So you can do 

```js
PF('yourtablewidget').collapseAllRows();
```

That will collapse all rows but there is no equivalent `expandAllRows()` so you may have to write a little JS code to do it.
